modules/standard/Regexp.chpl

```diff
@@ -408,37 +411,34 @@ class BadRegexpError : Error {
 }
 
 /*
-   Compile a regular expression. If the optional error argument is provided,
-   this routine will return an error code if compilation failed. Otherwise, it
-   will halt with an error message.
+   Compile a regular expression. This routine will throw a
+   class:`BadRegexpError` if compilation failed.
 
-   :arg pattern: the string regular expression to compile.
-                 See :ref:`regular-expression-syntax` for details. Note that
-                 you may have to escape backslashes. For example, to
+   :arg pattern: the regular expression to compile. This argument can be string
+                 or bytes. See :ref:`regular-expression-syntax` for details.
+                 Note that you may have to escape backslashes. For example, to
```

Comment:
I know this is unrelated to your PR, but while updating this documentation could you please emphasize in it that raw string literals exist? The docs here predate the raw string literals and regexps are an improtant motivating example.